### PR TITLE
[embedded] Disable Objective-C interop in embedded Swift

### PR DESCRIPTION
**Pull request by kubamracek**

### CODE REVIEW

**Comment by kubamracek** [comment]
@swift-ci please test

**Comment by kubamracek** [comment]
@swift-ci please test

**Comment by kubamracek** [comment]
@swift-ci please test

File: lib/Frontend/CompilerInvocation.cpp [hunk 1]
```diff
@@ -1095,7 +1095,7 @@ static bool ParseLangArgs(LangOptions &Opts, ArgList &Args,
 
   Opts.EnableObjCInterop =
       Args.hasFlag(OPT_enable_objc_interop, OPT_disable_objc_interop,
-                   Target.isOSDarwin());
+                   Target.isOSDarwin() && !Opts.hasFeature(Feature::Embedded));
```
Comment (zoecarver) [inline lib/Frontend/CompilerInvocation.cpp:1098]: Do we ever care about the non-darwin case explicitly enabling it with `enable-objc-interop`?

**Review by zoecarver — APPROVED** [review]
Makes sense! Thanks :)

**Comment by kubamracek** [comment]
@swift-ci please test

**Comment by kubamracek** [comment]
@swift-ci please test

**Comment by kubamracek** [comment]
@swift-ci please test

**Pull request merged**